File: jactor-client/src/main/java/com/github/jactor/rises/client/dto/GuestBookDto.java
package com.github.jactor.rises.client.dto;

import java.io.Serializable;
import java.util.Set;

public class GuestBookDto extends PersistentDto implements Serializable {
    private Set<GuestBookEntryDto> entries;
    private String title;
    private UserDto user;

    public GuestBookDto() {
        // empty, use setters
    }

    GuestBookDto(GuestBookDto guestBookDto) {
        super(guestBookDto);
        entries = guestBookDto.getEntries();
        title = guestBookDto.getTitle();
        user = guestBookDto.getUser();
    }

    public Set<GuestBookEntryDto> getEntries() {
        return entries;
    }

    public String getTitle() {
        return title;
    }

    public UserDto getUser() {
        return user;
    }

    public void setEntries(Set<GuestBookEntryDto> entries) {
        this.entries = entries;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setUser(UserDto user) {
        this.user = user;
    }
}

File: src/main/kotlin/com/github/jactor/persistence/dto/GuestBookDto.kt
package com.github.jactor.persistence.dto

data class GuestBookDto(
    override val persistentDto: PersistentDto = PersistentDto(),
    var entries: Set<GuestBookEntryDto> = emptySet(),
    var title: String? = null,
    var userInternal: UserInternalDto? = null
) : PersistentData(persistentDto) {
    constructor(
        persistentDto: PersistentDto, guestBook: GuestBookDto
    ) : this(
        persistentDto = persistentDto,
        entries = guestBook.entries,
        title = guestBook.title,
        userInternal = guestBook.userInternal
    )
}
